syntax = "proto3";
package com.example.grpc;

message HelloRequest {
  string name = 1;
  repeated string hobbies = 2;
}

message HelloResponse {
  string greeting = 1;
}

service GreetingService {
  rpc greetingServiceNonStream(HelloRequest) returns (HelloResponse);
  rpc greetingServiceServerStream(HelloRequest) returns (stream HelloResponse);
  rpc greetingServiceClientStream(stream HelloRequest) returns (HelloResponse);
  rpc greetingServiceBidirectionalStream(stream HelloRequest) returns (stream HelloResponse);
}